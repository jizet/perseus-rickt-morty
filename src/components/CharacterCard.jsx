import React from "react";
import { Link } from "wouter";

import styled from "styled-components";

const Card = styled.div`
  display: flex;
  flex-direction: row;
  width: 400px;
  height: 100px;
  margin: 10px;
  border-radius: 15px;
  background-color: white;
  overflow: hidden;
  cursor: pointer;
`;

const Info = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  color: #3c3e44;
  padding-left: 16px;
  flex-direction: column;
`;

const Img = styled.img`
  height: 100%;
`;

export const CharacterCard = ({
  id,
  name,
  status,
  origin,
  avatar,
  species,
}) => {
  return (
    <Link to={`/character/${id}`}>
      <Card>
        <Img src={avatar} alt="character-avatar" />
        <Info>
          <div>
            <b>Name:</b> {name}
          </div>
          <div>
            <b>Status:</b> {status}
          </div>
          <div>
            <b>Origin:</b> {origin}
          </div>
          <div>
            <b>Species:</b> {species}
          </div>
        </Info>
      </Card>
    </Link>
  );
};
