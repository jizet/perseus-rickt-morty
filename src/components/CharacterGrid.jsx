import React from "react";

import styled from "styled-components";
import { CharacterCard } from "components/CharacterCard";

const Container = styled.div`
  display: flex;
  width: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
`;

const GridContainer = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: flex-start;
  flex-direction: row;
  flex-wrap: wrap;
`;

export const CharacterGrid = ({ characters }) => {
  return (
    <Container>
      <GridContainer>
        {characters.map((character) => (
          <CharacterCard
            key={character.id}
            id={character.id}
            name={character.name}
            status={character.status}
            origin={character.origin.name}
            avatar={character.image}
            species={character.species}
          />
        ))}
      </GridContainer>
    </Container>
  );
};
