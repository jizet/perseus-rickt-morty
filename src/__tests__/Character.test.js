import React from "react";
import { CharacterCard } from "components/CharacterCard";
import renderer from "react-test-renderer";
import { characterMock } from "testUtils";

test("Character with data", () => {
  const component = renderer.create(
    <CharacterCard
      id={characterMock.id}
      name={characterMock.name}
      status={characterMock.status}
      origin={characterMock.origin.name}
      avatar={characterMock.image}
      species={characterMock.species}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
