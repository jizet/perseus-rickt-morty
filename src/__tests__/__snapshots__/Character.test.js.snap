// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character with data 1`] = `
<div
  className="sc-AxjAm kxXXrQ"
  href="/character/1"
  onClick={[Function]}
  to={null}
>
  <img
    alt="character-avatar"
    className="sc-AxiKw dKbEvk"
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <div
    className="sc-AxirZ iJAKen"
  >
    <div>
      <b>
        Name:
      </b>
       
      Rick Sanchez
    </div>
    <div>
      <b>
        Status:
      </b>
       
      Alive
    </div>
    <div>
      <b>
        Origin:
      </b>
       
      Earth (C-137)
    </div>
    <div>
      <b>
        Species:
      </b>
       
      Human
    </div>
  </div>
</div>
`;
